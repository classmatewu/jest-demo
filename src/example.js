const https = require('https')


const multiply = (a, b) => {
  return a * b
}

const fetchData = () => {
  return new Promise((resolve, reject) => {
    // const options = {
    //   hostname: 'opendata.baidu.com',
    //   port: 443,
    //   path: '/api.php?query=0.0.0.0&co=&resource_id=6006&oe=utf8',
    //   method: 'GET'
    // }
    const url = "https://opendata.baidu.com/api.php?query=0.0.0.0&co=&resource_id=6006&oe=utf8"
    
    const req = https.request(url, res => {
      console.log(`状态码: ${res.statusCode}`)
    
      res.on('data', d => {
        process.stdout.write(d)
        resolve(JSON.parse(d.toString()))
      })
    })

    req.on('error', error => {
      console.error(error)
      resolve(error)
    })
    
    req.end()
  })
}

export {
  multiply,
  fetchData
}