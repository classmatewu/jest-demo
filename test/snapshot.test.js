/**
 * 快照测试：通过保存快照，然后前后对比快照的方式，来测试一些不变的数据（配置、UI（html、css等）等）
 */

 import { getConfig } from '../src/example'

 // 第一次运行后，会在当前目录下生成 __snapshots__ 文件夹，连即保存了各个快照测试样例保存的快照，后续再 test 就会进行前后对比
 test('测试配置文件有没有更改', () => {
   expect(getConfig()).toMatchSnapshot() // 保存快照
 })