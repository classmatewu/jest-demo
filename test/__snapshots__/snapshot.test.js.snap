// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试配置文件有没有更改 1`] = `
{
  "port": 8080,
  "url": "http://localhost:3000",
}
`;
